hugr-core/src/extension/prelude/array.rs

```diff
@@ -179,6 +184,10 @@ impl MakeOpDef for ArrayOpDef {
     fn description(&self) -> String {
         match self {
             ArrayOpDef::new_array => "Create a new array from elements",
+            ArrayOpDef::new_uninitialized_array => {
+                "Create a new array with uninitialised elements. \
+                Reading such an unitialised element is undefined behaviour."
```

Comment:
Do we really want to have undefined behaviour in hugr? Perhaps it should panic instead (though that would require storing a "poison" value I suppose).

Reply:
I think guaranteeing a panic would be too much overhead. An alternative would be to pass a default value to the op and renaming it to `array_repeat` or something like that.

For linear arrays, this default would need to be provided via a function that is called for each element

Reply:
Yeah that could work. May add some runtime overhead (constructing values that are then replaced), but at least some of that could be eliminated by compiler optimizations.